### PR TITLE
[WIP] US 2018 sprint accessibility

### PR DESCRIPTION
**Pull request by raq929**
This begins work on the page navigation part of https://github.com/wagtail/wagtail/issues/4199#issuecomment-358640129

It wraps the main content of the admin in a `main` button, and changes a `footer` id in the nav menu to a more semantic `menu-footer`. This will allow future work on making footer tags more accessible. (I started working on this, so hopefully it's coming soon!)

Many thanks to @marteki for helping!

### CODE REVIEW

File: wagtail/admin/templates/wagtailadmin/base.html [hunk 1]
```diff
@@ -38,8 +38,9 @@
             </div>
 
             <div id="nav-toggle" class="nav-toggle icon text-replace">{% trans "Menu" %}</div>
-
-            {% block content %}{% endblock %}
+            <main>
```
Comment (jonnyscholes) [inline wagtail/admin/templates/wagtailadmin/base.html]: It is [still advised](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/main#Browser_compatibility) to include `role="main"` for IE11 which is still a supported browser. Probably worth including :)

File: wagtail/admin/static_src/wagtailadmin/js/core.js [hunk 2]
```diff
@@ -100,7 +100,7 @@ $(function() {
         $('.nav-wrapper').css('min-height', $(window).height());
         $('.nav-main').each(function() {
             var thisHeight = $(this).height();
```
Comment (jjanssen) [inline wagtail/admin/static_src/wagtailadmin/js/core.js]: Although it does not fit the scope of the pull-request, I noticed the by the change in the next-line the`thisHeight` variable is not being used. Is this something you could cleanup in this pull-request as well? :smile:

File: wagtail/admin/templates/wagtailadmin/shared/main_nav.html [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
+        <li class="footer" id="menu-footer">
             <div class="account" id="account-settings" title="{% trans 'Edit your account' %}">
                 <span class="avatar square avatar-on-dark">
                     <img src="{% avatar_url request.user size=50 %}" />
                 </span>
                 <em class="icon icon-arrow-up-after">{{ request.user.first_name|default:request.user.get_username }}</em>
             </div>
 
-            <ul class="footer-submenu">
+            <ul class="footer-submenu" aria-label="Account">
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/shared/main_nav.html]: What is the reasoning for this change? I see the menu items underneath both have their own labels – does it make sense to have this as a labelled section of the menu?

Also should be using `trans` for translations, `{% trans "Account" %}`.

File: wagtail/admin/templates/wagtailadmin/shared/main_nav.html [hunk 4]
```diff
@@ -1,18 +1,18 @@
 {% load wagtailadmin_tags %}
 {% load i18n %}
-<nav class="nav-main">
+<nav role="navigation" class="nav-main" aria-label="{% trans "Main navigation" %}">
     <ul>
         {{ menu_html }}
 
-        <li class="footer" id="footer">
+        <li class="footer" id="menu-footer">
```
Comment (thibaudcolas) [inline wagtail/admin/templates/wagtailadmin/shared/main_nav.html]: As far as I know the semantics of ids does not have an impact on assistive technology – could you explain this change further?

File: wagtail/admin/templates/wagtailadmin/base.html (hunk 1 above)
Comment (thibaudcolas) [reply wagtail/admin/templates/wagtailadmin/base.html]: Yes, this should still be a `div` tag with the appropriate role to maintain browser support without extra JS.

**Comment by thibaudcolas** [comment]
This is now replaced by #5290.

**Pull request closed without merging**